student.rb

```diff
@@ -1,9 +1,17 @@
 require './person'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(classroom, age, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission)
     @classroom = classroom
+    add_classroom(classroom)
+  end
+
+  def add_classroom(classroom)
+    @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
```

Comment:
**OPTIONAL** - Good job attempting to create a custom setter for the Student class, but you have not implemented it properly. I suggest you do this:
```ruby
def add_classroom(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
end
```